### ISSUE TITLE
[PATCH] correctly pass test name as fourth arg in t/op/print.t

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#13537]
Migrated from [rt.perl.org#121031](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=121031) (status was 'resolved')

Searchable as RT121031$

**Comment by p5pRT** [comment]
### From @FROGGS

Hi\, the test in t/op/print\.t called fresh\_perl\_is\(\) wrong\-ish\. It
expects four args and was called with three by passing the test name as
compiler flags\.

Patch attached\.

**Comment by p5pRT** [comment]
### From @FROGGS

<details><summary>0001-correctly-pass-test-name-as-fourth-arg.patch</summary>

```diff
From 7a10bdaf0ebfd4b1a832370af97483f529e3440d Mon Sep 17 00:00:00 2001
From: Tobias Leich <email@froggs.de>
Date: Thu, 26 Dec 2013 13:06:01 +0100
Subject: [PATCH] correctly pass test name as fourth arg

fresh_perl_is expects compiler flags as the third arg. Pass an empty
string if we don't have any. This way the test name shows up instead
of the ode we run.
---
 t/op/print.t | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/t/op/print.t b/t/op/print.t
index 5e508f6..7dafe4f 100644
--- a/t/op/print.t
+++ b/t/op/print.t
@@ -6,9 +6,9 @@ BEGIN {
 
 plan(3);
 
-fresh_perl_is('$_ = qq{OK\n}; print;', "OK\n",
+fresh_perl_is('$_ = qq{OK\n}; print;', "OK\n", '',
               'print without arguments outputs $_');
-fresh_perl_is('$_ = qq{OK\n}; print STDOUT;', "OK\n",
+fresh_perl_is('$_ = qq{OK\n}; print STDOUT;', "OK\n", '',
               'print with only a filehandle outputs $_');
 SKIP: {
     skip_if_miniperl('no dynamic loading of PerlIO::scalar in miniperl');
-- 
1.8.3.2


```
</details>

**Comment by p5pRT** [comment]
### From @jkeenan

On Sat Jan 18 12&#8203;:24&#8203;:49 2014\, FROGGS\.de wrote&#8203;:
> Hi\, the test in t/op/print\.t called fresh\_perl\_is\(\) wrong\-ish\. It
> expects four args and was called with three by passing the test name as
> compiler flags\.
> 
> Patch attached\.

Thanks for noting this problem\.  However\, when I look at other places in t/op/ where fresh\_perl\_is\(\) is called and on the code for \_fresh\_perl\(\) in t/test\.pl\, I think the test should be corrected to use an empty hash reference rather than an empty string\.

Also\, in your commit message&#8203;: s/ode/code/

Thank you very much\.
Jim Keenan

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @tonycoz

On Sun Jan 19 15&#8203;:59&#8203;:59 2014\, jkeenan wrote&#8203;:
> On Sat Jan 18 12&#8203;:24&#8203;:49 2014\, FROGGS\.de wrote&#8203;:
> > Hi\, the test in t/op/print\.t called fresh\_perl\_is\(\) wrong\-ish\. It
> > expects four args and was called with three by passing the test name
> > as
> > compiler flags\.
> >
> > Patch attached\.
> 
> Thanks for noting this problem\.  However\, when I look at other places
> in t/op/ where fresh\_perl\_is\(\) is called and on the code for
> \_fresh\_perl\(\) in t/test\.pl\, I think the test should be corrected to
> use an empty hash reference rather than an empty string\.
> 
> Also\, in your commit message&#8203;: s/ode/code/

Passing "" is unsafe since it means %\{""\} will be modified\.

Passing undef should be safe\, since $runperl\_args will be autovivified\, but perhaps we shouldn't be depending on that at this level of testing \(given the "Baby perl" note at the top of test\.pl\)

I discussed this with FROGGS in IRC\, I'm producing a better patch\, and fixing all of the other similar bugs\.

Tony

**Comment by p5pRT** [comment]
### From @tonycoz

On Sun Jan 19 16&#8203;:23&#8203;:34 2014\, tonyc wrote&#8203;:
> Passing "" is unsafe since it means %\{""\} will be modified\.
> 
> Passing undef should be safe\, since $runperl\_args will be
> autovivified\, but perhaps we shouldn't be depending on that at this
> level of testing \(given the "Baby perl" note at the top of test\.pl\)
> 
> I discussed this with FROGGS in IRC\, I'm producing a better patch\, and
> fixing all of the other similar bugs\.

I've applied a variant of FROGG's patch as de482fd5eff91548320228250421b402b7faaa35\, passing \{\} instead of ""\.

I also temp patched \_fresh\_perl\(\) to pick up other calls that don't supply undef or a hashref and fixed the problems that revealed in 20e5bab43efd0e449d0741f5c5a278e7e20ee9dc\.

Tony

**Comment by p5pRT** [comment]
@tonycoz - Status changed from 'open' to 'resolved'
